### PR TITLE
Dict Score Modifiers API tests

### PR DESCRIPTION
**Pull request by RaynorChavez**
* **Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes/features)
- [x] Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Feature

* **What is the current behavior?** (You can also link to an open issue here)
No tests for dictionary score modifiers

* **What is the new behavior (if this is a feature change)?**
Added tests for dictionary score modifiers implemented in https://github.com/marqo-ai/marqo/pull/851

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

* **Other information**:

### CODE REVIEW

File: tests/api_tests/test_dict_score_modifiers.py [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+                }
+            },
+            {
+                "indexName": cls.unstructured_index_name,
+                "type": "unstructured",
+                "model": "open_clip/ViT-B-32/laion2b_s34b_b79k"
+            }
+        ])
+
+        cls.indexes_to_delete = [cls.structured_index_name, cls.unstructured_index_name]
```
Comment (farshidz) [inline tests/api_tests/test_dict_score_modifiers.py:50]: I don't think you need this. Doesn't the base class add the indexes to this list when you call create?

**Review by farshidz — CHANGES_REQUESTED** [review]
Please also test:
* More than one value in maps
* Different combinations e.g., document has standard score modifier fields and maps, and use some or all of them in search
* Cases where we had a bug (all score modifiers specified missing in doc, or some of them missing). Test cases where value isn't in map, and also when map is completely empty
* Negative cases e.g. score modifier specified doesn't exist in map, also when map itself doesn't exist 
*

File: tests/api_tests/test_dict_score_modifiers.py (hunk 1 above)
Comment (RaynorChavez) [reply tests/api_tests/test_dict_score_modifiers.py:50]: From: https://github.com/marqo-ai/marqo-api-tests/blob/mainline/tests/marqo_test.py 

I think this line is necessary

**Comment by RaynorChavez** [comment]
- [x] More than one value in maps
- [x] Different combinations e.g., document has standard score modifier fields and maps, and use some or all of them in search
- Cases where we had a bug (
- [x] all score modifiers specified missing in doc, or 
- [x] some of them missing). 
- [x] Test cases where value isn't in map, and also when 
- [x] map is completely empty
- Negative cases e.g. score modifier specified doesn't exist in map (seems redundant with the others above?), 
- [x] also when map itself doesn't exist

**Review by farshidz — APPROVED** [review]

**Pull request merged**